proposed/2023/Add-dotnet-search.md

```diff
@@ -0,0 +1,107 @@
+| `--help` | Show command help and usage information |
+| `--verbosity` | Display the amount of details in the output: normal, quiet, detailed. |
+|||
+
+#### **Algorithm**
+
+1. Read and parse arguments from user
+2. Use the nuget.exe Search API to look for the specific term in the provided source. The search filter will be composed of the `--source`, `[Search term]`, `--prerelease` values.
+3. Based on `--verbosity` value, provide with a list of packages with various verbosity
+
+- Verbosity
+  - **Quiet** : Each line would look as follows :
+
+            >[Package Name] | [Latest Package Version]
+  - **Normal** :
+
+                >[Package Name] | [Latest Package Version] | [Amount of Downloads]
+                <Description of the Package(short form)>
+  - **Detailed** :
+
+                >[Package Name] | [Latest Package Version] | [Amount of Downloads]
+                Deprecated : True/False | Vulnerable : True/False
+                <Description of the Package(Readme file)>
+                License URL : [URL]
+
+#### **New Option `-exact-match`**
+
+- This new option will allow for users to be able to search and have only exact matches as an output.
+- For example if a user uses `dotnet package search NuGet.CommandLine --verbosity Quiet`
+
+        >NuGet.CommandLine | 6.7.0
+        >NuGet.CommandLine.XPlat | 6.7.0
+        >NuGet.Commands | 6.7.0
+        >NuGet.exe | 3.4.3
+        >NuGet.Bootstrapper | 2.6.0
+        >CommandLineParser20 | 2.0.0
+        >NuGet.VerifyMicrosoftPackage | 1.0.0
+        >NuGet.for.MSBuild | 2.1.0 ...
+
+- Using ``dotnet package search NuGet.CommandLine --exact-match --verbosity Quiet`` on the other side will have the following output
+
+         >NuGet.CommandLine | 6.7.0
+
+## Drawbacks
+
+<!-- Why should we not do this? -->
+
+## Rationale and alternatives
+
+<!-- Why is this the best design compared to other designs? -->
+<!-- What other designs have been considered and why weren't they chosen? -->
+<!-- What is the impact of not doing this? -->
+In nuget.exe there is `nuget search` command which does the same thing. However, customers would like this functionality to be available in dotnet.
+
+## Prior Art
+
+<!-- What prior art, both good and bad are related to this proposal? -->
+<!-- Do other features exist in other ecosystems and what experience have their community had? -->
+<!-- What lessons from other communities can we learn from? -->
+<!-- Are there any resources that are relevant to this proposal? -->
+
+## Unresolved Questions
+
+<!-- What parts of the proposal do you expect to resolve before this gets accepted? -->
+<!-- What parts of the proposal need to be resolved before the proposal is stabilized? -->
+<!-- What related issues would you consider out of scope for this proposal but can be addressed in the future? -->
+
+## Future Possibilities
+
+<!-- What future possibilities can you think of that this proposal would help with? -->
+Make the search command bigger and vast
```

Comment:
IMO this is too vague. If you don't have specific examples of what feature could be added, I'd leave this section empty